DatadogTrace/Sources/OpenTelemetry/OTelSpan.swift

```diff
@@ -0,0 +1,162 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+* This product includes software developed at Datadog (https://www.datadoghq.com/).
+* Copyright 2019-Present Datadog, Inc.
+*/
+
+import Foundation
+import OpenTelemetryApi
+
+internal enum DatadogTagKeys: String {
+    case spanKind = "span.kind"
+    case errorMessage = "error.Message"
+}
+
+internal class OTelSpan: OpenTelemetryApi.Span {
+    private var _status: OpenTelemetryApi.Status
+    private var _name: String
+
+    var attributes: [String: OpenTelemetryApi.AttributeValue]
+    let context: OpenTelemetryApi.SpanContext
+    var kind: OpenTelemetryApi.SpanKind
+    let nestedSpan: DDSpan
```

Comment:
nit/ "nested" in tracing lingo might be mistaken with "child". I feel `ddSpan` might work better here and in other places where we need to follow this pattern. It's my personal preference so feel free to ignore 😉.